web/Makefile

```diff
@@ -0,0 +1,135 @@
+DEFAULT: dev
+
+SERVER_PORT=8000
+REPO_ROOT:=$(shell git rev-parse --show-toplevel)
+
+##
+# Section: Tasks (Phony targets)
+##
+
+APPS=abstreet fifteen_min osm_viewer santa widgetry_demo 
+
+.PHONY: server clean run dev release $(APPS)
+.PRECIOUS: src/%/wasm_pkg
+
+TSC=npx tsc --lib dom,es2020 --module es2020 --target es6 --strict --noEmitOnError --outDir build
+
+server:
+	python -m http.server $(SERVER_PORT) --directory build/dist
+
+clean:
+	rm -fr build
+	rm -fr src/*/wasm_pkg
+
+run: dev
+	echo "Make sure server is up and running first!"
+	open http://localhost:$(SERVER_PORT)
+
+# Build all of our apps
+build: $(APPS) build/dist/index.html
+
+# Build all of our apps for distribution (optimized)
+release: export WASM_PACK_FLAGS=--release
+# note: `clean` is because we don't know if any cached wasm files were optimized
+# Once you've compiled the optimized wasm this way, you can skip `clean` and just run `make build`, since the remaining output is identical for dev vs release
+release: clean build
+
+dev: export WASM_PACK_FLAGS=--dev
+dev: build build/dist/data
+
+# Symlink in data for dev builds
+build/dist/data:
+	ln -sf ../../../data build/dist/data
+
+##
+# Section: Shared Deps and Templates
+##
+
+build/dist/%.html: src/web_root/*.html
+	# just copy everything... we could do something more nuanced
+	cp -r src/web_root/* build/dist
+
+# Produce the wasm package via our wasm-pack script
+# To skip optimization:
+#     WASM_PACK_FLAGS="--dev" make
+src/%/wasm_pkg:
+	bin/build-wasm $*
+
+# simply copies over the wasm pkg we 
+# built to be able to compile our typescript
+build/dist/%/wasm_pkg: src/%/wasm_pkg
+	mkdir -p $(dir $@)
+	cp -r "${<}" "${@}"
+
+# Concatenate the target crate's js file with 
```

Comment:
Just curious, do makefiles have really short word wrap standards for comments?

Reply:
nah, I think I just failed to rewrap after editing it. I've rewrapped to something more reasonable.